Shopify/cli
# [Themes] Refactor `fetchStoreThemes` function to prepare for public themes API

### WHY are these changes introduced?

Fixes https://github.com/Shopify/develop-advanced-edits/issues/347

### WHAT is this pull request doing?
Allows users to provide a store `FQDN` and `password` generated by the `Theme Access` app to list themes on a shop.

### How to test your changes?
- Hover to see JSDoc definitions
- More extensive testing instructions can be found [here](https://github.com/Shopify/cli/pull/4518)

### Measuring impact

How do we know this change was effective? Please choose one:


### Checklist
